server/repo-mgr.c

```diff
@@ -255,6 +260,68 @@ should_ignore_file(const char *filename, void *data)
     return FALSE;
 }
 
+static gboolean
+collect_repo_id (SeafDBRow *row, void *data);
+
+static int
+scan_trash (void *data)
+{
+    GList *repo_ids = NULL;
+    SeafRepoManager *mgr = seaf->repo_mgr;
+    gint64 expire_time = time(NULL) - mgr->priv->trash_expire_interval;
+    char *sql = "SELECT repo_id FROM RepoTrash WHERE del_time <= ?";
+
+    int ret = seaf_db_statement_foreach_row (seaf->db, sql,
+                                             collect_repo_id, &repo_ids,
+                                             1, "int64", expire_time);
+    if (ret < 0) {
+        seaf_warning ("Get expired repo from trash failed.");
+        string_list_free (repo_ids);
+        return TRUE;
+    }
+
+    GList *iter;
+    char *repo_id;
+    for (iter=repo_ids; iter; iter=iter->next) {
+        repo_id = iter->data;
+        ret = seaf_repo_manager_del_repo_from_trash (mgr, repo_id, NULL);
+        if (ret < 0)
+            break;
+    }
+
+    string_list_free (repo_ids);
+
+    return TRUE;
+}
+
+static void
+init_scan_trash_timer (SeafRepoManagerPriv *priv, GKeyFile *config)
+{
+    int scan_days;
+    int expire_days;
+    GError *error = NULL;
+
+    scan_days = g_key_file_get_integer (config,
+                                        "RepoTrash", "scan_days",
```

Comment:
"library_trash" is better. Need to be consistent with the naming format of other options.